### PR TITLE
Emit warning when SPNEGO auth used for an XA-capable datasource and no recovery auth is defined

### PR DESCRIPTION
**Pull request by aguibert**

### CODE REVIEW

**Comment by aguibert** [comment]
#build

File: dev/com.ibm.ws.jca.cm/resources/com/ibm/ws/j2c/resources/J2CAMessages.nlsprops
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.jca.cm/resources/com/ibm/ws/j2c/resources/J2CAMessages.nlsprops:325]: Suggested update

>A connection request for the {0} XA resource was made by using SPNEGO authentication, but no recovery authentication data alias was configured. The XA resource is not able to participate in automated XA recovery unless a recovery authentication data alias is configured.

File: dev/com.ibm.ws.jca.cm/resources/com/ibm/ws/j2c/resources/J2CAMessages.nlsprops
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.jca.cm/resources/com/ibm/ws/j2c/resources/J2CAMessages.nlsprops:326]: Update

>As a result, SPNEGO authentication can not be used for automatic XA recovery.

To

>As a result, SPNEGO authentication can't be used for automatic XA recovery.

File: dev/com.ibm.ws.jca.cm/resources/com/ibm/ws/j2c/resources/J2CAMessages.nlsprops
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.jca.cm/resources/com/ibm/ws/j2c/resources/J2CAMessages.nlsprops:327]: Suggested update

>Configure a recovery authentication data alias on the XA resource.

**Review by Charlotte-Holt — COMMENTED** [review]
@aguibert Just a few quick ID comments. Let me know if you have questions. Thx!

File: dev/com.ibm.ws.jca.cm/resources/com/ibm/ws/j2c/resources/J2CAMessages.nlsprops
Comment (aguibert) [reply dev/com.ibm.ws.jca.cm/resources/com/ibm/ws/j2c/resources/J2CAMessages.nlsprops:327]: Thanks for the comments @Charlotte-Holt. I'll update your other 2 suggestions, but I'm going to decline this one because I think "auth data" is more clear to the user, because in config what the user types in is actually `<authData>` or `xxxAuthDataRef`

**Comment by aguibert** [comment]
#build

File: dev/com.ibm.ws.jca.cm/resources/com/ibm/ws/j2c/resources/J2CAMessages.nlsprops
Comment (Charlotte-Holt) [reply dev/com.ibm.ws.jca.cm/resources/com/ibm/ws/j2c/resources/J2CAMessages.nlsprops:327]: Got it. That makes sense. Thanks, @aguibert.

**Comment by aguibert** [comment]
build is good, ready to merge once approved

#run-libby

**Review by mswatosh — APPROVED** [review]

**Pull request merged**
